lib/publiccloud/basetest.pm

```diff
@@ -116,19 +131,18 @@ sub finalize {
     diag('Public Cloud finalize: $flags->{publiccloud_multi_module}=' . $flags->{publiccloud_multi_module}) if ($flags->{publiccloud_multi_module});
     diag('Public Cloud finalize: $flags->{fatal}=' . $flags->{fatal}) if ($flags->{fatal});
     diag('Public Cloud finalize: $self->{result}=' . $self->{result}) if ($self->{result});
-    diag('Public Cloud finalize: $self->{run_args}=' . $self->{run_args}) if ($self->{run_args});
 
-    if ($self->{run_args}) {
-        if ($self->{run_args}->{my_instance}) {
-            my $dumpable_instance = Storable::dclone($self->{run_args}->{my_instance});
-            $dumpable_instance->{provider}->{provider_client}->{credentials_file_content} = '******';
-            diag('Public Cloud finalize: $self->{run_args}->{my_instance}=' . Dumper($dumpable_instance));
-        }
-        if ($self->{run_args}->{my_provider}) {
-            my $dumpable_provider = Storable::dclone($self->{run_args}->{my_provider});
-            $dumpable_provider->{provider_client}->{credentials_file_content} = '******';
-            diag('Public Cloud finalize: $self->{run_args}->{my_provider}=' . Dumper($dumpable_provider));
-        }
+    if ($self->instance) {
+        my $dumpable_instance = Storable::dclone($self->instance);
+        # We sensor the credentials as they're secret
+        $dumpable_instance->{provider}->{provider_client}->{credentials_file_content} = '******';
```

Comment:
I think the comment can be made redundant:
```suggestion
        $dumpable_instance->{provider}->{provider_client}->{credentials_file_content} = CENSORED_SECRET;
```
and at the top of the module:
```perl
use constant CENSORED_SECRET => '******';
```
(same below in line 143)